### PR TITLE
Document ERC721ReceiverComponent

### LINKED ISSUE #913: Add docs for ERC721Receiver component
Right now only its interface is documented

image

### PR DESCRIPTION
**Pull request by ericnordelo**
Fixes #913   

This PR adds the missing documentation for the ERC721ReceiverComponent, and also
fixes the name of the mixin implementation.

#### PR Checklist

- [x] Tests
- [x] Documentation
- [x] Added entry to CHANGELOG.md 
- [ ] Tried the feature on a public network

### CODE REVIEW

**Comment by ericnordelo** [comment]
> I think we can update [Creating a token receiver contract](https://github.com/ericnordelo/cairo-contracts/blob/feat/document-erc721receiver-%23913/docs/modules/ROOT/pages/erc721.adoc?plain=1#L214-L223) to use ERC721ReceiverMixinImpl as well to clean up the snippet. WDYT?

The only issue I see with the approach is inconsistency. For example, the ERC721 in the same page is not using mixins, and I think we should update that as well. Let's let it for a different issue if you agree.

File: docs/modules/ROOT/pages/api/erc721.adoc [hunk 1]
```diff
@@ -620,6 +622,70 @@ Interface for contracts that support receiving `safe_transfer_from` transfers.
...
+--
+.InternalImpl
+* xref:#ERC721ReceiverComponent-initializer[`++initializer(self)++`]
+--
+
+==== Embeddable functions
+
+[.contract-item]
+[[ERC721ReceiverComponent-on_erc721_received]]
+==== `[.contract-item-name]#++on_erc721_received++#++(operator: ContractAddress, from: ContractAddress, token_id: u256, data Span<felt252>) -> felt252++` [.item-kind]#external#
```
Comment (andrew-fleming) [inline docs/modules/ROOT/pages/api/erc721.adoc:671]:
```suggestion
==== `[.contract-item-name]#++on_erc721_received++#++(self: @ContractState, operator: ContractAddress, from: ContractAddress, token_id: u256, data Span<felt252>) -> felt252++` [.item-kind]#external#
```

File: docs/modules/ROOT/pages/api/erc721.adoc [hunk 2]
```diff
@@ -620,6 +622,70 @@ Interface for contracts that support receiving `safe_transfer_from` transfers.
...
+
+Returns the `IERC721Receiver` interface ID.
+
+[.contract-item]
+[[ERC721ReceiverComponent-onERC721Received]]
+==== `[.contract-item-name]#++onERC721Received++#++(operator: ContractAddress, from: ContractAddress, token_id: u256, data Span<felt252>) -> felt252++` [.item-kind]#external#
```
Comment (andrew-fleming) [inline docs/modules/ROOT/pages/api/erc721.adoc:677]:
```suggestion
==== `[.contract-item-name]#++onERC721Received++#++(self: @ContractState, operator: ContractAddress, from: ContractAddress, token_id: u256, data Span<felt252>) -> felt252++` [.item-kind]#external#
```

**Review by andrew-fleming — APPROVED** [review]
I left a couple more small suggestions and they need to be applied in #941 as well. This looks good to me otherwise

**Review by martriay — APPROVED** [review]

**Pull request merged**